File: src/main/java/com/uiptv/server/api/json/HttpIptvM3u8Server.java
package com.uiptv.server.api.json;

import com.sun.net.httpserver.HttpExchange;
import com.sun.net.httpserver.HttpHandler;
import com.uiptv.service.M3U8PublicationService;

import java.io.IOException;

import static com.uiptv.util.ServerUtils.generateM3u8Response;

public class HttpIptvM3u8Server implements HttpHandler {
    @Override
    public void handle(HttpExchange ex) throws IOException {
        String response = M3U8PublicationService.getInstance().getPublishedM3u8();
        String path = ex.getRequestURI().getPath();
        String filename = "iptv.m3u8";
        if (path.endsWith(".m3u")) {
            filename = "iptv.m3u";
        }
        generateM3u8Response(ex, response, filename);
    }
}

File: server/src/main/kotlin/com/uiptv/server/api/json/HttpIptvM3u8Server.kt
package com.uiptv.server.api.json

import com.sun.net.httpserver.HttpExchange
import com.sun.net.httpserver.HttpHandler
import com.uiptv.service.PlaylistExportService
import com.uiptv.util.ServerUtils.generateM3u8Response
import java.io.IOException

class HttpIptvM3u8Server : HttpHandler {
    private val playlistExportService = PlaylistExportService()

    data class PlaylistDocument(val content: String, val fileName: String)

    @Throws(IOException::class)
    override fun handle(ex: HttpExchange) {
        val document = buildPublishedPlaylist(ex.requestHeaders.getFirst("Host"), ex.requestURI.path)
        generateM3u8Response(ex, document.content, document.fileName)
    }

    fun buildPublishedPlaylist(host: String?, requestPath: String?): PlaylistDocument =
        playlistExportService.buildPublishedPlaylist(host, requestPath).let { PlaylistDocument(it.content, it.fileName) }
}
